src/main/java/org/broadinstitute/barclay/argparser/CommandLineException.java

```diff
@@ -39,6 +41,36 @@ public BadArgumentValue(String message) {
         }
     }
 
+    public static class OutOfRangeArgumentValue extends BadArgumentValue {
+        private static final long serialVersionUID = 0L;
+
+        public OutOfRangeArgumentValue(final String argName, final double minValue, final double maxValue, final Object value) {
+            super(argName, getValueString(value), getMessage(minValue, maxValue, value instanceof Integer));
+        }
+
+        // to handle null values
+        private static String getValueString(final Object value) {
+            return (value == null) ? "null" : value.toString();
+        }
+
+        // get the message for the values, correctly formatted
+        private static String getMessage(final double minValue, final double maxValue, final boolean asInt) {
+            final boolean outMinValue = minValue != Double.NEGATIVE_INFINITY;
+            final boolean outMaxValue = maxValue != Double.POSITIVE_INFINITY;
+            final DoubleFunction<String> toString = (asInt) ? Double::toString : v -> Integer.toString((int) Math.rint(v));
+            if (outMinValue && outMaxValue) {
+                return String.format("allowed range [%s, %s].", toString.apply(minValue), toString.apply(maxValue));
+            } else if (outMinValue) {
+                return String.format("minimum allowed value %s", toString.apply(minValue));
+            } else if (outMaxValue) {
+                return String.format("maximum allowed value %s", toString.apply(maxValue));
+            }
+            // this should never be reached
+            return "unknown";
```

Comment:
@magicDGS I know you asked about this in the last round but I never commented. I think we should throw here since it shouldn't ever happen.